### PR TITLE
Upgrade rn version to 0.62.2

### PR DESCRIPTION
**Pull request by aguscha333**
Upgrade the base so it uses react-native 0.62.2.

This PR follows the diff changes in the upgrade helper, [here is a link](https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.62.2) with the diff between versions. We went from version 0.61.5 to 0.62.2.

Although version 0.63.x is already out, it is fairly new so it is best to wait till issues arise and are fixed.

In case you want to see what is new in react-native 0.62 [here](https://medium.com/@kaissaroj/whats-new-in-react-native-0-62-5a9305d65d) is an article that talks about that but the main update is support for a new debugging tool called Flipper that is very promising and also new devtools that can be used as always from the browser.

Aside from that, `redux-flipper` integration was added so redux timeline can be accessed from this new tool.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1 +1,3 @@
 *.pbxproj -text
+# specific for windows script files
+*.bat text eol=crlf
```
Comment (agustito37) [inline .gitattributes:3]: EOF

File: android/app/build.gradle [hunk 2]
```diff
@@ -1,9 +1,9 @@
 apply plugin: "com.android.application"
 
 project.ext.envConfigFiles = [
-    dev: ".env",
-    staging: ".env.staging",
-    prod: ".env.prod",
+        dev: ".env",
```
Comment (agustito37) [inline android/app/build.gradle:4]: Why this changed?

File: android/app/src/main/java/com/reactnativebase/MainApplication.java [hunk 3]
```diff
@@ -69,4 +75,4 @@ private static void initializeFlipper(Context context) {
       }
     }
   }
-}
+}
```
Comment (agustito37) [inline android/app/src/main/java/com/reactnativebase/MainApplication.java:78]: EOF

File: ios/Podfile [hunk 4]
```diff
@@ -40,35 +94,46 @@ target 'ReactNativeBase' do
...
 
 target 'ReactNativeBase-Staging' do
-    # Pods for ReactNativeBase
-    base_pods
-
-
-    use_native_modules!
+  # Pods for ReactNativeBase
+  base_pods
+
```
Comment (agustito37) [inline ios/Podfile:109]: Weird double line break

File: ios/Podfile [hunk 5]
```diff
@@ -40,35 +94,46 @@ target 'ReactNativeBase' do
...
     # Pods for testing
   end
-
+  
 end
+
+
+post_install do |installer|
+  flipper_post_install(installer)
+end
```
Comment (agustito37) [inline ios/Podfile:139]: EOF

File: package.json [hunk 6]
```diff
@@ -28,26 +28,30 @@
     "@react-navigation/native": "^5.0.0",
     "@react-navigation/stack": "^5.0.0",
     "@rootstrap/redux-tools": "^0.4.2",
+    "add": "^2.0.6",
```
Comment (bernabe9) [inline package.json:31]: What is this library for? 🤔

File: package.json [hunk 7]
```diff
@@ -28,26 +28,30 @@
...
     "humps": "2.0.1",
     "immer": "5.0.0",
     "lodash": "4.17.15",
     "prop-types": "15.7.2",
-    "react": "16.9.0",
-    "react-native": "0.61.5",
+    "react": "16.11.0",
+    "react-native": "0.62.2",
     "react-native-config": "0.12.0",
+    "react-native-flipper": "^0.50.0",
```
Comment (bernabe9) [inline package.json:41]: Should this library be a devDependency?

File: package.json [hunk 8]
```diff
@@ -28,26 +28,30 @@
...
     "react-native-screens": "^2.0.0-beta.2",
     "react-redux": "7.1.1",
     "redux": "4.0.4",
+    "redux-flipper": "^1.3.1",
     "redux-form": "8.2.3",
     "redux-persist": "^6.0.0",
     "reselect": "4.0.0",
-    "validate.js": "0.12.0"
+    "validate.js": "0.12.0",
+    "yarn": "^1.22.4"
```
Comment (bernabe9) [inline package.json:54]: 🤔

File: android/app/src/debug/java/com/reactnativebase/ReactNativeFlipper.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+                      }
+                    });
+              }
+            });
+      } else {
+        client.addPlugin(new FrescoFlipperPlugin());
+      }
+    }
+  }
+}
```
Comment (fernandatoledo) [inline android/app/src/debug/java/com/reactnativebase/ReactNativeFlipper.java:67]: Another newline missing

File: ios/Podfile (hunk 4 above)
Comment (aguscha333) [reply ios/Podfile:108]: fixed

File: package.json (hunk 7 above)
Comment (aguscha333) [reply package.json]: In the instructions it didn't specify that you should put it as a dev dependency, I tried anyway and it seems to be working.

**Comment by aguscha333** [comment]
@bernabe9 fixed

**Pull request merged**